src/Adapter/Presenter/Product/ProductLazyArray.php

```diff
@@ -149,13 +149,18 @@ public function getAttributes()
     }
 
     /**
+     * We check the catalog mode separately, because shouldShowPrice is used in
+     * other methods of this class for fine tuning display not related to catalog
+     * mode. However when this getter is used somewhere in a template we always hide
+     * the price in catalog mode.
+     *
      * @arrayAccess
      *
      * @return bool
      */
     public function getShowPrice()
     {
-        return $this->shouldShowPrice($this->settings, $this->product);
+        return $this->shouldShowPrice($this->settings, $this->product) && !$this->settings->catalog_mode;
```

Comment:
`return !$this->settings->catalog_mode && $this->shouldShowPrice($this->settings, $this->product);` will be faster.

Reply:
Shouldn't this be done in `shouldShowPrice()` instead?